# Task 3: A/B Hypothesis Testing

**Statistical Validation of Risk Drivers**

Testing key hypotheses about risk differences across provinces, zip codes, and gender to support risk-based pricing and segmentation strategy.

### Objective

Statistically validate or reject the following hypotheses:
- H₀: There are no risk differences across provinces
- H₀: There is no significant margin difference between zip codes
- H₀: There is no significant risk difference between Women and Men

In [13]:
import importlib
import src.hypothesis_tests 

importlib.reload(src.hypothesis_tests)

<module 'src.hypothesis_tests' from 'c:\\projects\\week-3\\week-3-insurance-analytics\\src\\hypothesis_tests.py'>

In [8]:
# 10 Academy Week 3 Challenge
import sys
import os

sys.path.append(os.path.abspath('..'))

import pandas as pd
import numpy as np
from src.hypothesis_tests import *

# Load data
df = load_data()
print(f"Data loaded successfully. Shape: {df.shape}")

Data loaded successfully. Shape: (1000098, 57)


## Hypothesis Testing Results

In [14]:
results = run_hypothesis_tests(df)

Running Hypothesis Tests for Task 3...


CHI-SQUARED TEST: Claim Frequency - Province
Comparing: Gauteng vs Western Cape
Chi2 Statistic : 56.0874
p-value        : 0.000000
Decision       : Reject H0 (Significant Difference)

CHI-SQUARED TEST: Claim Frequency - Gender
Comparing: Male vs Female
Chi2 Statistic : 0.0037
p-value        : 0.951464
Decision       : Fail to reject H0

T-TEST: Margin Difference - Province
Comparing: Gauteng vs Western Cape
p-value  : 0.163596
Decision : Fail to reject H0


## Business Interpretations & Recommendations

### 1. Provinces (Claim Frequency)
- **Test**: Chi-Squared Test (Gauteng vs Western Cape)
- **p-value**: 0.000000
- **Decision**: **Reject H₀**

**Insight**: There is a **statistically significant difference** in claim frequency between Gauteng and Western Cape. Gauteng has notably higher risk.

**Recommendation**: Implement **regional risk adjustments** — higher premiums or stricter underwriting in Gauteng, while Western Cape (and similar low-risk provinces) could be targeted for growth.

---

### 2. Gender (Claim Frequency)
- **Test**: Chi-Squared Test (Male vs Female)
- **p-value**: 0.951464
- **Decision**: **Fail to reject H₀**

**Insight**: No statistically significant difference in claim frequency between Male and Female policyholders.

**Recommendation**: Gender may **not** be a strong risk driver in this portfolio. Avoid using gender-based pricing to prevent potential regulatory issues.

---

### 3. Provinces (Margin)
- **Test**: Independent T-Test (Gauteng vs Western Cape)
- **p-value**: 0.163596
- **Decision**: **Fail to reject H₀**

**Insight**: Although Gauteng has higher claim frequency, the difference in **Margin** (profit per policy) is not statistically significant.

**Recommendation**: Focus more on **Claim Frequency** than Margin when designing risk segments.

---

### Overall Strategic Recommendation

The analysis provides strong evidence for **geographic risk segmentation**. ACIS should consider:
- Increasing premiums or reducing exposure in high-risk provinces like **Gauteng**.
- Offering more competitive premiums in lower-risk provinces to attract new customers.
- Avoid gender-based differentiation due to lack of statistical significance.